src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/ParameterSymbolReferenceFinder.cs

```diff
@@ -121,17 +121,21 @@ protected override bool CanFind(IParameterSymbol symbol)
                 }
             }
 
-            return results ?? SpecializedCollections.EmptyEnumerable<SymbolAndProjectId>();
+            return results == null
+                ? ImmutableArray<SymbolAndProjectId>.Empty
+                : results.ToImmutable();
         }
 
-        private IEnumerable<SymbolAndProjectId> CascadeBetweenAnonymousFunctionParameters(
+        private ImmutableArray<SymbolAndProjectId> CascadeBetweenAnonymousFunctionParameters(
```

Comment:
FAR doesn't benefit from lazyness becuase it always evaluates everything.  I was just very IEnumerable happy in my youth.  But i've found myself distrusting of it due to hidden unintended lazyness.  in codebases like far (with a large amount of async work), it's very easy to mess something up.  i.e. you Task.Run an async call to offload hte work to another thread.  It returns an enumerable.  You then traverse that enumerable on your original thread, and now all the work actually happens.   Whoopsie :)